sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Common/AnalyzerConfiguration.cs

```diff
@@ -0,0 +1,96 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Common
+{
+    public class AnalyzerConfiguration
+    {
+        private static readonly Lazy<HotspotConfiguration> hotspotConfiguration =
+            new Lazy<HotspotConfiguration>(() => new HotspotConfiguration());
```

Comment:
It's probably minor but as the code of the `HotspotConfiguration` relies on the class being a singleton I would hardcode the singleton in the class (even if you have the property on this outer class). It would make it obvious and avoid (potential later) wrong change.